### DISCUSSION TITLE
Store settings in mysql or S3

### DISCUSSION THREAD

**Discussion by goellner** [aerni/statamic-advanced-seo#76] [Ideas]
I am working on a multi tenant statamic app and my only issue atm is that no SEO plugin supports storing the settings in eloquent or even in a s3 bucket. 

The ideal SEO plugin would store those settings in a custom table in the SQL database. The alternative would be to store the settings in  an S3 bucket. 

The only issues are the settings that get stored in `content/advanced-seo`, everything else works perfectly when using the statamic eloquent driver (https://github.com/statamic/eloquent-driver).

**Comment by goellner** [comment]
if your plugin could save that as a global set, that would be fine too. think thats even the easiest solution

**Comment by aerni** [comment]
Do you have any concerns about storing the Advanced SEO defaults as files? 

Performance wise, it shouldn't make a big difference compared to storing them in a DB. There are very few files that will be saved in Statamic's Stache, which should be able to handle them pretty easily.

Saving the defaults as a global set is not an option.

**Comment by goellner** [comment]
I have one Statamic Laravel app and I have all the data for all tenants in their own sql db. So I can't really use the current approach of the package, because they all share one install. Has nothing to do with performance really, just a separation of data.

**Comment by aerni** [comment]
Gotcha. I might add support for an Eloquent driver in the future. Have been thinking about it before.

**Comment by aerni** [comment]
Eloquent support will be added with https://github.com/aerni/statamic-advanced-seo/pull/186.
